### ISSUE TITLE
`View.IsOverridden` doesn't work

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#2156]
I wrote these unit tests (ViewTests.cs) to prove it.

```csharp
		[Fact, AutoInitShutdown]
		public void IsOverridden_False_IfNotOverriden ()
		{
			var view = new DerivedView () { Text = "DerivedView does not override MouseEvent", Width = 10, Height = 10 };
			
			Assert.False (view.IsOverridden (view, "MouseEvent"));
		}

		[Fact, AutoInitShutdown]
		public void IsOverridden_True_IfOverriden ()
		{
			var view = new Button () { Text = "Button overrides MouseEvent", Width = 10, Height = 10 };

			Assert.True (view.IsOverridden (view, "MouseEvent"));
		}
```

<img width="365" alt="image" src="https://user-images.githubusercontent.com/585482/199355799-afda1cca-ea1d-4e9c-846b-7b0fc9980c37.png">

@BDisp - Where'd you get this code from? 

```csharp
		/// <summary>
		/// Check if the <paramref name="method"/> is overridden in the <paramref name="view"/>.
		/// </summary>
		/// <param name="view">The view.</param>
		/// <param name="method">The method name.</param>
		/// <returns><see langword="true"/> if it's overridden, <see langword="false"/> otherwise.</returns>
		public bool IsOverridden (View view, string method)
		{
			Type t = view.GetType ();
			MethodInfo m = t.GetMethod (method);

			return (m.DeclaringType == t || m.ReflectedType == t) && m.GetBaseDefinition ().DeclaringType == typeof (Responder);
		}
```

**Comment by tig** [comment]
In fact, even the positive case is broken in some cases!

```csharp
		[Fact, AutoInitShutdown]
		public void IsOverridden_True_IfOverriden ()
		{
			var view = new ScrollBarView () { Text = "ScrollBarView overrides both MouseEvent and Redraw", Width = 10, Height = 10 };

			Assert.True (view.IsOverridden (view, "MouseEvent"));
			Assert.True (view.IsOverridden (view, "Redraw"));
		}
````

`ScrollBarView` definitely overrides both MouseEvent and Redraw. But the 2nd Assert here fails:

<img width="363" alt="image" src="https://user-images.githubusercontent.com/585482/199356856-6a402d41-1608-4489-ba77-1b57bfb1d662.png">

**Comment by BDisp** [comment]
Well the test is correct because the virtual `MouseEvent` method belongs to the `Responder` class and is overridden in the `View` class. So. it's natural the test is true. It's a disadvantage to not belong to the `View`, like the `Redraw`. Not everything is perfect :-)

**Comment by tig** [comment]
This passes the tests:

```csharp
		/// <summary>
		/// Check if the <paramref name="method"/> is overridden in the <paramref name="view"/>.
		/// </summary>
		/// <param name="view">The view.</param>
		/// <param name="method">The method name.</param>
		/// <returns><see langword="true"/> if it's overridden, <see langword="false"/> otherwise.</returns>
		public static bool IsOverridden (View view, string method)
		{
			MethodInfo m = view.GetType ().GetMethod (method, 
				BindingFlags.Instance 
				| BindingFlags.Public 
				| BindingFlags.NonPublic 
				| BindingFlags.DeclaredOnly);
			if (m == null) {
				return false;
			}
			return m.GetBaseDefinition ().DeclaringType != m.DeclaringType;
		}
```

**Comment by BDisp** [comment]
> @BDisp - Where'd you get this code from?

That was the code where is were used in the `ScrollView` and I moved it to the `View` class.

**Comment by BDisp** [comment]
Your fix seems to work very well. Go ahead, thanks.

**Comment by BDisp** [comment]
Your fix also fix the #2140. Only replace the `ComboBoxTests` `HideDropdownListOnClick_True_Highlight_Current_Item` in the line 827 with this. Because now the `SelectedItem` default value is -1 then no item are selected on the `ListView` of the `ComboBox`.

```cs
			TestHelpers.AssertDriverColorsAre (@"
000000
22222
22222
22222", attributes);

```
